modules/imgcodecs/include/opencv2/imgcodecs.hpp

```diff
@@ -53,6 +53,52 @@
   @}
 */
 
+/* duplicate of "ImreadModes" enumeration for better compatibility with OpenCV 3.x */
+enum
+{
+/* 8bit, color or not */
+    CV_LOAD_IMAGE_UNCHANGED  =-1,
```

Comment:
Can we keep them in other header files? For example, `imgcodecs/c_api_compatibility.hpp` or current `imgcodecs.h` (under `__cplusplus`).

Main purpose of this to avoid using of these deprecated symbols:
- in OpenCV code itself
- in bindings parsers
- don't bloat documentation